### PR TITLE
refactor: move limit query from state to urlParams

### LINKED ISSUE #2154: refactor: move limit query from `useState` to `urlSearchParams` across application
### Type of feature

🧑‍💻 Refactor

### Current behavior

Similar to what i started in https://github.com/open-sauced/app/pull/2047, we need to also store the limit filter state in the browsers url query. This is also related to https://github.com/open-sauced/app/issues/712.

### Suggested solution

refactor the useState approach to using the browsers urlQuery params

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR refactors how the limit query state is stored across the application. Now we have the limit in the url of the pages

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #2154

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: lib/hooks/api/useContributorList.ts
Comment (nickytonline) [inline lib/hooks/api/useContributorList.ts:36]: Refactor (nit): Prefer a separate pull request for the changes for range as it is not related to the current refactor.

File: lib/hooks/api/useContributorList.ts
Comment (nickytonline) [inline lib/hooks/api/useContributorList.ts:40]: Refactor (blocking): The `defaultLimit` already has a default value of `"10"`, so `?? "10"` is note required.

```suggestion
  query.append("limit", defaultLimit);
```

File: pages/lists/[listId]/activity.tsx
Comment (nickytonline) [inline pages/lists/[listId]/activity.tsx:34]: Naming (nit): No need to specify `Server` in the type name.

```suggestion
export type FilterParams = {
```

Also, `FilterParams` is used on the list activity page but also elsewhere. Consider moving the type elsewhere and import it into the list activity page.

**Review by nickytonline — CHANGES_REQUESTED** [review]

File: pages/lists/[listId]/activity.tsx
Comment (nickytonline) [inline pages/lists/[listId]/activity.tsx:47]: Refactor (blocking): `RawLimit` should be camel case as it's a JavaScript variable and not a type.

```suggestion
  const { listId, range: rawRange, limit: rawLimit } = ctx.params as ServerFilterParams;
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
This is looking good @OgDev-01, just some change requests.

**Comment by brandonroberts** [comment]
@OgDev-01 I think the changes are good, but using strings as something that should be a number seems off. I know that's how they are in the URL, but I think we should convert them to a numbers at some point.

**Comment by nickytonline** [comment]
> @OgDev-01 I think the changes are good, but using strings as something that should be a number seems off. I know that's how they are in the URL, but I think we should convert them to a numbers at some point.

Yeah it seemed off to me too. I mentioned it in the changes for the range changes PR as well although it wasn't blocking the PR. Agree with Brandon here even if they become strings in the end.

We can revisit it for `range` after.

File: components/atoms/Select/limit-select.tsx
Comment (jpmcb) [reply components/atoms/Select/limit-select.tsx:15]: @nickytonline there are a few places within the API where we default to some set limit. Example:

https://github.com/open-sauced/api/blob/24892059033ee8e9855517d887475cdb54f8681f/src/user/dtos/top-users.dto.ts#L25-L36

But as far as I can tell, these numbers are arbitrary. The upper limit on many of these are intended to prevent full table scans.

File: pages/lists/[listId]/contributors.tsx
Comment (jpmcb) [inline pages/lists/[listId]/contributors.tsx:78]: Looks like do a cast here we cast this number to a string to be consumed in the URL? Like brandon mentioned, I'm also curious if there's a way to prevent storing strings as numbers (especially if it'll be converted to a number eventually and cast back to a string here)

Comment (nickytonline) [reply pages/lists/[listId]/contributors.tsx:78]: This one is king of inevitable because it comes from the router via the querystring. It's type is `string | string[] | undefined`, so in this case, this would need to become most likely this

`defaultLimit: Number(limit as string),`

**Comment by OgDev-01** [comment]
Hey @open-sauced/engineering, I've reverted the limit type to number and converted it to string in the appropriate areas. this is ready for a second review 🍕

**Review by jpmcb — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**
